pynicotine/gtkgui/widgets/textentry.py

```diff
@@ -160,7 +160,7 @@ def add_completion(self, item):
 
     def remove_completion(self, item):
 
-        iterator = self.completions.pop(item)
+        iterator = self.completions.pop(item, None)
 
         if iterator is not None:
```

Comment:
> I would prefer to continue raising a KeyError here. We should never hit it unless something has gone wrong elsewhere.

In that case, iterator can never be None so this condition can be removed @mathiascode

Reply:
It's None when the completion dropdown is disabled.